File: backend/src/main/java/com/project2/domain/member/entity/Follows.java
package com.project2.domain.member.entity;

import com.project2.global.entity.BaseTime;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Entity
public class Follows extends BaseTime {
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Long id;

	@ManyToOne
	@JoinColumn(nullable = false)
	private Member follower;

	@ManyToOne
	@JoinColumn(nullable = false)
	private Member following;
}

File: backend/src/main/java/com/project2/domain/member/entity/Follows.kt
package com.project2.domain.member.entity

import com.project2.global.entity.BaseTime
import jakarta.persistence.*
import lombok.AllArgsConstructor
import lombok.Getter
import lombok.NoArgsConstructor


@Entity
@NoArgsConstructor
class Follows : BaseTime {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long? = null

    @ManyToOne
    @JoinColumn(nullable = false)
    var follower: Member? = null

    @ManyToOne
    @JoinColumn(nullable = false)
    var following: Member? = null

    constructor(id: Long?, follower: Member?, following: Member?) {
        this.id = id
        this.follower = follower
        this.following = following
    }

}
